lib/travis/api/app.rb

```diff
@@ -106,6 +106,11 @@ def initialize
         use Travis::Api::App::Middleware::Skylight
         use(Rack::Config) { |env| env['metriks.request.start'] ||= Time.now.utc }
 
+        if ENV['HONEYCOMB_ENABLED_FOR_DYNOS']&.split(' ')&.include?(ENV['DYNO'])
+          Travis.logger.info 'honeycomb enabled'
+          use Travis::Api::App::Middleware::Honeycomb, writekey: ENV['HONEYCOMB_WRITEKEY'], dataset: ENV['HONEYCOMB_DATASET']
```

Comment:
this is indeed a concern. 👍  we've enabled each one of these on a different dyno in order to avoid that type of issue.